application/backend/src/main/resources/apps/typerefinery/components/graphs/barchart/clientlibs/functions.js

```diff
@@ -0,0 +1,204 @@
+            componentConfig = {
+                ...componentConfig,
+                ...DEFAULT_BAR_CHART_DATA
+            }
+            console.log("componentConfig-2",componentConfig)
+            var componentDataPath = $component.getAttribute("data-resource-path");
+            console.log("path",componentDataPath)
+            const ctx =  document.getElementById(`${componentDataPath}-barChart`).getContext("2d");
+
+
+            // Linear background for the chart.
+        
+            // Plugin to update the canvas Background.
+            const plugin = {
+              id: "customCanvasBackgroundColor",
+              beforeDraw: (chart, args, options) => {
+                const { ctx } = chart;
+                ctx.save();
+                ctx.globalCompositeOperation = "destination-over";
+                ctx.fillStyle ="#99ffff";
+                ctx.fillRect(0, 0, chart.width, chart.height);
+                ctx.restore();
+              },
+            };          
+            // Line chart
+            const chartInstance = new Chart(ctx, {
+              type: "bar",
+              data: {
+                labels: data.labels || componentConfig.labels,
+                datasets: [
+                  {
+                    axis: "y",
+                    data: data.chartData || componentConfig.chartData,
+                    fill: false,
+                    backgroundColor:"blue",
+                    borderColor: [
+                      'rgb(255, 99, 132)',
+                      'rgb(255, 159, 64)',
+                      'rgb(255, 205, 86)',
+                      'rgb(75, 192, 192)',
+                      'rgb(54, 162, 235)',
+                      'rgb(153, 102, 255)',
+                      'rgb(201, 203, 207)'
+                    ],
+                    borderWidth: 1,
+                  },
+                ],
+              },
+              options: {
+                indexAxis: "y",
+                plugins: {
+                  legend: {
+                    display: false,
+                  },
+                },
+                scales: {
+                  x: {
+                    grid: {
+                      display: false,
+                    },
+                    ticks: {
+                      color:"#5D7183",
+                    },
+                  },
+                  y: {
+                    grid: {
+                      color: "#5D7183",
+                    },
+                    ticks: {
+                      color:"#5D7183"
+                    },
+                  },
+                },
+                // customCanvasBackgroundColor: {
+                //   color: canvasBackgroundColor,
+                // },
+                interaction: {
+                  intersect: false,
+                },
+                radius: 0,
+              },
+              plugins: [plugin],
+            });
+          
+            
```

Comment:
Make them chart values dynamic.
I see everything is static.

Please refer to linechart

Reply:
Done